### PR TITLE
min operator

### PR DESCRIPTION
**Pull request by rjhdby**
added reactiveX min operator

### CODE REVIEW

**Comment by davidwdan** [comment]
This PR needs functional tests

File: lib/Rx/Operator/MinOperator.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public function __invoke(
+        ObservableInterface $observable,
+        ObserverInterface $observer,
+        SchedulerInterface $scheduler = null
+    ) {
+        $previousMin = null;
+        $comparing   = false;
+
+        return $observable->subscribe(new CallbackObserver(
+                                          function ($x) use (&$comparing, &$previousMin, $observer) {
```
Comment (davidwdan) [inline lib/Rx/Operator/MinOperator.php]: Indentation is off.  It should be 12 spaces here.

**Comment by rjhdby** [comment]
@davidwdan What kind of functional tests? Can you explain please?
https://github.com/rjhdby/RxPHP/blob/minOperator/demo/min/min.php
https://github.com/rjhdby/RxPHP/blob/minOperator/demo/min/min-with-comparer.php
Is not enough?

**Comment by rjhdby** [comment]
@davidwdan 
How I understand after read MaxTest, in functional tests first call of onNext is skipped when check results?

```
public function max_number_Some()
    {
        $xs = $this->createHotObservable([
            onNext(150, 1),  <-- If I change it to onNext(150, 5), test still passed normaly
            onNext(210, 3),
            onNext(220, 4),
            onNext(230, 2),
            onCompleted(250)
        ]);

        $results = $this->scheduler->startWithCreate(function () use ($xs) {
            return $xs->max();
        });

        $this->assertMessages([
            onNext(250, 4),      <-- Valid result is 4
            onCompleted(250)
        ], $results->getMessages());
    }
```

Where I can read about right organisation of tests for RxPHP?

**Comment by davidwdan** [comment]
@rjhdby when using the test scheduler, subscriptions don't happen until [200](https://github.com/ReactiveX/RxPHP/blob/master/lib/Rx/Testing/TestScheduler.php#L11), so anything before 200 never gets emitted.

**Comment by davidwdan** [comment]
:+1: LGTM

**Pull request merged**